pytests/helpers/session.py

```diff
@@ -461,20 +467,26 @@ def _wait_for_remaining_output(self):
         for thread in self._output_capture_threads:
             thread.join()
 
-    def common_setup(self, path, run_as, breakpoints=()):
+    def common_setup(self, path, starttype, run_as, breakpoints=(), backchannel=False):
```

Comment:
I'm thinking that at this point there doesn't appear to be a clear distinction between `prepare_to_run` and `common_setup`. Also, we have a lot of inconsistency here in that some things are passed as arguments, while others (like `expected_returncode`) are object attributes.

I suggest replacing both with a single method - let's call it something like `initialize()` - that takes no arguments at all. All session parameters are configured consistently via object attributes. So, you'd do something like this:
```py
debug_session.start_method = ('launch',)
debug_session.target = 'script.py'
debug_session.run_as = 'file'
debug_session.breakpoints = [...]
debug_session.initialize()
```
or:
```py
debug_session.start_method = ('attach', 'socket')
debug_session.target = 'module_name'
debug_session.run_as = 'module'
debug_session.backchannel = True
debug_session.initialize()
```
As a shortcut, we can make it `initialize(**kwargs)`, which would then be treated simply as setting the corresponding attributes. Thus:
```py
debug_session.initialize(
     start_method = ('attach', 'socket'),
     run_as = 'module',
     target = 'module_name',
     backchannel = True,
)
```

Reply:
Another idea is to combine `target` and `run_as` in the same manner as `start_method`. E.g.:
```py
debug_session.target = ('file', 'script.py')
debug_session.target = ('module', 'foo.bar')
debug_session.target = ('code', 'print("ok")')
```
and allow the second item in the tuple to be a function instead of a str, in which case `@pyfile` is automatically applied to it, and then the module name is derived from the filename, or raw code extracted from the file, depending on the first item. This would make it possible to write tests like `test_run` in a more generic way, so instead of this:
```py
    @pyfile
    def code_to_debug():
       ...

    if run_as == 'file':
        debug_session.prepare_to_run(filename=code_to_debug, backchannel=True)
    elif run_as == 'module':
        debug_session.add_file_to_pythonpath(code_to_debug)
        debug_session.prepare_to_run(module='code_to_debug', backchannel=True)
    elif run_as == 'code':
        with open(code_to_debug, 'r') as f:
            code = f.read()
        debug_session.prepare_to_run(code=code, backchannel=True)
    else:
        pytest.fail()
```
you can simply do:
```py
    def code_to_debug():
        ...
    debug_session.initialize(target=(run_as, code_to_debug))
```
and then `initialize` will contain the equivalent of the conditional above.